src/test/ledger/Invariants_test.cpp

```diff
@@ -215,6 +229,176 @@ class Invariants_test : public beast::unit_test::suite
             });
     }
 
+    void
+    testXRPBalanceCheck(bool enabled)
+    {
+        using namespace test::jtx;
+        testcase << "checks " << (enabled ? "enabled" : "disabled") <<
+            " - XRP balance checks";
+
+        doInvariantCheck (enabled,
+            {{ "Cannot return non-native STAmount as XRPAmount" }},
```

Comment:
hmm .. I'm not sure how to trigger that exc since none of the current invariants invoke `iou()`. Did you have something specific in mind?